### PR TITLE
docs: updates and improvements to integration section

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

This section was screaming for a facelift...

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

## Summary by CodeRabbit

- **Documentation**
  - Updated integration guide for IBC module, detailing steps for configuring Keepers, setting up routers, managing modules, and handling ABCI ordering.

### CODE REVIEW

**Review by colin-axner — APPROVED** [review]
Thank you @crodriguezvega!

File: docs/docs/01-ibc/02-integration.md [hunk 1]
```diff
@@ -19,64 +19,28 @@ send fungible token transfers to other chains.
...
-- Add modules to `Begin/EndBlockers` and `InitGenesis`
-- Update the module `SimulationManager` to enable simulations
+- [Define additional `Keeper` fields for the new modules on the `App` type](#add-application-fields-to-app).
+- [Add the module's `StoreKey`s and initialize their `Keeper`s](#configure-the-keepers).
+- [Set up IBC router and add route for the `transfer` module](#register-module-routes-in-the-ibc-router).
+- [Grant permissions to `transfer`'s `ModuleAccount`](#module-account-permissions).
+- [Add the modules to the module `Manager`](#module-manager-and-simulationmanager).
+- [Update the module `SimulationManager` to enable simulations](#module-manager-and-simulationmanager).
+- [Integrate light client modules (e.g. `07-tendermint`)](#integrating-light-clients).
+- [Add modules to `Begin/EndBlockers` and `InitGenesis`](#Application ABCI ordering).
```
Comment (damiannolan) [inline docs/docs/01-ibc/02-integration.md:29]:
```suggestion
- [Add modules to `Begin/EndBlockers` and `InitGenesis`](#application-abci-ordering).
```

File: docs/docs/01-ibc/02-integration.md [hunk 2]
```diff
@@ -147,84 +139,194 @@ Currently, a `Router` is static so it must be initialized and set correctly on a
...
+)
+```
+
+### Application ABCI ordering
+
+One addition from IBC is the concept of `HistoricalEntries` which are stored on the Cosmos SDK `x/staking` module.
 Each entry contains the historical information for the `Header` and `ValidatorSet` of this chain which is stored
 at each height during the `BeginBlock` call. The historical info is required to introspect the
 past historical info at any given height in order to verify the light client `ConsensusState` during the
 connection handshake.
```
Comment (damiannolan) [inline docs/docs/01-ibc/02-integration.md:275]:
```suggestion
Each entry contains the historical information for the `Header` and `ValidatorSet` of this chain which is stored
at each height during the `BeginBlock` call. The `HistoricalInfo` is required to introspect a blockchain's prior state at a given height in order to verify the light client `ConsensusState` during the
connection handshake. Note that this construct is specific to `07-tendermint` light client implementations.
```

File: docs/docs/01-ibc/02-integration.md [hunk 3]
```diff
@@ -147,84 +139,194 @@ Currently, a `Router` is static so it must be initialized and set correctly on a
...
+
+  // register light clients on IBC
+  // highlight-next-line
++ ibctm.NewAppModule(tmLightClientModule),
+)
+```
+
+### Application ABCI ordering
+
+One addition from IBC is the concept of `HistoricalEntries` which are stored on the Cosmos SDK `x/staking` module.
```
Comment (damiannolan) [inline docs/docs/01-ibc/02-integration.md:271]:
```suggestion
One addition from IBC is the concept of `HistoricalInfo` which is stored in the Cosmos SDK `x/staking` module. The number of records stored by `x/staking` is controlled by the `HistoricalEntries` parameter which stores `HistoricalInfo` on a per height basis.
```

**Review by damiannolan — APPROVED** [review]
Really nice improvements, thanks @crodriguezvega!

**Pull request merged**